src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/ApplyVQSRIntegrationTest.java

```diff
@@ -0,0 +1,151 @@
+package org.broadinstitute.hellbender.tools.walkers.vqsr;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFCodec;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+import org.broadinstitute.hellbender.utils.variant.GATKVariant;
+import org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapter;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ApplyVQSRIntegrationTest extends CommandLineProgramTest {
+
+    @Override
+    public String getTestedClassName() {
+        return ApplyVQSR.class.getSimpleName();
+    }
+
+    public String getToolTestDataDir(){
+        return publicTestDir + "org/broadinstitute/hellbender/tools/VQSR/";
+    }
+
+    public String getLargeVQSRTestDataDir(){
+        return largeFileTestDir + "VQSR/";
+    }
+
+    @Test
+    public void testApplySNPRecalibration() throws IOException {
+        final String inputFile = getLargeVQSRTestDataDir() + "phase1.projectConsensus.chr20.1M-10M.raw.snps.vcf";
+
+        final IntegrationTestSpec spec = new IntegrationTestSpec(
+                " -L 20:1,000,000-10,000,000" +
+                    " --variant " + inputFile +
+                    " --lenient" +
+                    " --output %s" +
+                    " -mode SNP" +
+                    " -tranchesFile " + getLargeVQSRTestDataDir() + "expected/SNPTranches.txt" +
+                    // pass in the tranche file to match GATK3; though without a TS_FILTER_LEVEL
+                    // arg they aren't used
+                    " -recalFile " + getLargeVQSRTestDataDir() + "snpRecal.vcf",
+                Arrays.asList(getLargeVQSRTestDataDir() + "expected/snpApplyResult.vcf"));
```

Comment:
Document how the expected output(s) were generated in these tests -- eg., were they generated with GATK3? Did anything have to change between 3 and 4?